convert-hf-to-powerinfer-gguf.py

```diff
@@ -0,0 +1,601 @@
+#!/usr/bin/env python3
+
+from __future__ import annotations
+from abc import ABC, abstractmethod
+
+import argparse
+import contextlib
+import json
+import os
+import re
+import struct
+import sys
+from enum import IntEnum
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, ContextManager, Iterator, Optional, cast
+
+import numpy as np
+import torch
+import torch.nn as tnn
+
+if TYPE_CHECKING:
+    from torch import Tensor
+
+if "NO_LOCAL_GGUF" not in os.environ:
+    sys.path.insert(1, str(Path(__file__).parent / "gguf-py"))
+import gguf
+
+
+###### MODEL DEFINITIONS ######
+
+
+class SentencePieceTokenTypes(IntEnum):
+    NORMAL = 1
+    UNKNOWN = 2
+    CONTROL = 3
+    USER_DEFINED = 4
+    UNUSED = 5
+    BYTE = 6
+
+
+class ReluMLP(tnn.Module):
```

Comment:
Relus are an important part of their story because Relus allowed them for sparsity. See their [reddit post](https://www.reddit.com/r/LocalLLaMA/comments/18luk10/wait_llama_and_falcon_are_also_moe/):

> However, an interesting observation that LLM also have sparse activation due to ReLU function. Based on ReLU-based LLM([SparseLLM (SparseLLM) (huggingface.co)](https://huggingface.co/SparseLLM)), we implement a fast inference system, PowerInfer. ... We definitly find that only 20% neurons consistently contributes to the majority of activations!